src/App.tsx

```diff
@@ -17,18 +67,30 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                filterBy={filter}
+                setFilter={handleChangeFilter}
+                query={query}
+                setQuery={handleQueryChange}
+                cleaner={queryCleaner}
+              />
             </div>
 
+            {loading && <Loader />}
             <div className="block">
-              <Loader />
-              <TodoList />
+              <TodoList
```

Comment:
```suggestion
              loading  ? <Loader /> : <TodoList />
```

It might not be worth rendering the list at all while the data is being loaded